mycroft/api/__init__.py

```diff
@@ -26,6 +37,11 @@ def request(self, params):
                                     data=params.get("data"), json=json)
         return self.get_response(response)
 
+    def request(self, params):
+        self.check_token()
+        params["path"] = self.build_path(params)
```

Comment:
Move the initialization to inside the build process and return the final result instead